From 5838b6390353e4c34762828684cee90410e4f8b1 Mon Sep 17 00:00:00 2001
From: Russell Bryant <rbryant@redhat.com>
Date: Tue, 24 Apr 2012 12:24:39 -0400
Subject: [PATCH] Fix content type for qpid notifier.

Fix bug 980872.

This patch fixes a regression I introduced in
2d36facf14f4eb2742ba46274e04a73b5231aece.  In that patch, I adjusted the
content_type for messages sent with the qpid notifier to be
'application/json' to match a change that went into the kombu notifier.
Unfortunately, it's wrong.

I assumed based on the kombu change that notifications were being json
encoded before being passed into the notification driver.  That's not
the case.  The message is a dict.  So, just revert the change to set the
content_type and let Qpid encode the notification as 'amqp/map'.

(cherry picked from commit 5bed23cbc962d3c6503f0ff93e6d1e326efbd49d)

Change-Id: I8ba039612d9603377028ba72cb80ae89d675c741
---
 glance/notifier/notify_qpid.py     |    9 +++------
 glance/tests/unit/test_notifier.py |    2 +-
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/glance/notifier/notify_qpid.py b/glance/notifier/notify_qpid.py
index d3d5514..a0535a6 100644
--- a/glance/notifier/notify_qpid.py
+++ b/glance/notifier/notify_qpid.py
@@ -135,16 +135,13 @@ class QpidStrategy(strategy.Strategy):
         return self.session.sender(address)
 
     def warn(self, msg):
-        qpid_msg = qpid.messaging.Message(content=msg,
-                                          content_type='application/json')
+        qpid_msg = qpid.messaging.Message(content=msg)
         self.sender_warn.send(qpid_msg)
 
     def info(self, msg):
-        qpid_msg = qpid.messaging.Message(content=msg,
-                                          content_type='application/json')
+        qpid_msg = qpid.messaging.Message(content=msg)
         self.sender_info.send(qpid_msg)
 
     def error(self, msg):
-        qpid_msg = qpid.messaging.Message(content=msg,
-                                          content_type='application/json')
+        qpid_msg = qpid.messaging.Message(content=msg)
         self.sender_error.send(qpid_msg)
diff --git a/glance/tests/unit/test_notifier.py b/glance/tests/unit/test_notifier.py
index 0e7ff75..b952ee3 100644
--- a/glance/tests/unit/test_notifier.py
+++ b/glance/tests/unit/test_notifier.py
@@ -316,7 +316,7 @@ class TestQpidNotifier(unittest.TestCase):
         super(TestQpidNotifier, self).tearDown()
 
     def _test_notify(self, priority):
-        test_msg = json.dumps({'a': 'b'})
+        test_msg = {'a': 'b'}
 
         self.mock_connection = self.mocker.CreateMock(self.orig_connection)
         self.mock_session = self.mocker.CreateMock(self.orig_session)
